test/eeprom.py

```diff
@@ -0,0 +1,114 @@
+import tkinter as tk
+from datetime import date, datetime
+from tkinter import Button, Entry, Label
+
+import paramiko
+import pytest
+
+
+def save_to_eeprom(iio_uri, coarse, fine):
+    full_uri = iio_uri.split(":", 2)
+    if full_uri[0] != "ip":
+        pytest.skip("Tuning currently supported only for ip URIs")
+    ip = full_uri[1]
+
+    ssh_client = paramiko.SSHClient()
+    ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh_client.connect(ip, username="root", password="analog")
+    ssh_stdin, ssh_stdout, ssh_stderr = ssh_client.exec_command(
+        "find /sys/ -name eeprom"
+    )
+    eeprom_path = ssh_stdout.readline().rstrip("\n")
+
+    if ssh_stderr.channel.recv_exit_status() != 0:
+        raise paramiko.SSHException("find_eeprom command did not execute properly")
+
+    this_day = date.today()
+    if this_day.year <= 1995:
+        pytest.skip(
+            "System date and time not in order. PLease connect the device to internet to update."
+        )
+
+    sn = popup_txt()
```

Comment:
Can we add a check here to make sure the input is reasonable

Reply:
I added a check in the popup_txt() function, right before returning the value